# Our implemation of Sampling

In [1]:
import pandas as pd
import numpy as np

In [6]:
rain = pd.read_csv('./data/derekswork/DTW_prec.csv')

In [9]:
rain #That's a lot of data! If it was gigabytes, we could use just a sample for testing while developing.

DATE  PRCP
0       1/1/2001  0.00
1       1/2/2001  0.00
2       1/3/2001  0.00
3       1/4/2001  0.04
4       1/5/2001  0.14
...          ...   ...
6186  12/27/2017  0.00
6187  12/28/2017  0.00
6188  12/29/2017  0.00
6189  12/30/2017  0.00
6190  12/31/2017  0.00

[6191 rows x 2 columns]

In [16]:
sample = rain.sample(n=5)
sample
#Different sample each time this cell is run (sampled without replacement)

DATE  PRCP
2065   8/31/2006  0.00
1430   12/1/2004  0.32
2787   8/24/2008  0.00
2832   10/8/2008  0.14
4003  12/23/2011  0.00

In [22]:
sample_based_on_percent = rain.sample(frac=0.1, random_state = 69)
sample_based_on_percent

DATE  PRCP
1567  4/20/2005  0.09
4053  2/11/2012  0.00
1192   4/7/2004  0.00
948    8/7/2003  0.00
5188  3/28/2015  0.00
...         ...   ...
199   7/19/2001  0.00
4388  1/11/2013  0.69
2217  1/30/2007  0.01
3860   8/2/2011  0.10
4474   4/7/2013  0.00

[619 rows x 2 columns]

#### notice the above cell always has 619 values (if frac=0.1). As in, there's no variance, so this is more deterministic than a true random binomial sample.

#### Sometimes it's in our interest to sample WITH replacement, allowing a data point to be taken more than once through multiple draws, without being exhausted. This is how dice work, for example. (rolling a 1 doesn't exclude a 1 from being rolled next) 

#### In such a large dataset as this, the difference between the two methods is small, but still there.

In [23]:
sample_with_replacement = rain.sample(frac=0.1, replace=True, random_state=69)
sample_with_replacement

DATE  PRCP
4041   1/30/2012   0.0
2667   4/26/2008   0.0
1626   6/18/2005   0.0
439    3/16/2002   0.0
3633  12/18/2010   0.0
...          ...   ...
1456  12/27/2004   0.0
5515   2/21/2016   0.0
4341  11/25/2012   0.0
2377   7/11/2007   0.0
2043    8/9/2006   0.0

[619 rows x 2 columns]